AssertionError: cant parse version xyz

Hi,

One of our python projects uses the setuptools_scm tooling, but has started failing for versions > 2.1.0.

```
sphinx-latest runtests: commands[1] | mlx-warnings --sphinx --maxwarnings 7 --minwarnings 7 .tox/doc_html.log
  /home/travis/build/melexis/sphinx-traceability-extension$ /home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/bin/mlx-warnings --sphinx --maxwarnings 7 --minwarnings 7 .tox/doc_html.log 
/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/git.py:68: UserWarning: "/home/travis/build/melexis/sphinx-traceability-extension" is shallow and may cause errors
  warnings.warn('"%s" is shallow and may cause errors' % (wd.path,))
/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/version.py:92: UserWarning: tag 'V2.5.2' no version found
  warnings.warn("tag %r no version found" % (tag,))
Traceback (most recent call last):
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/bin/mlx-warnings", line 7, in <module>
    from mlx.warnings import main
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/mlx/warnings.py", line 16, in <module>
    __version__ = get_version()
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/__init__.py", line 135, in get_version
    parsed_version = _do_parse(config)
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/__init__.py", line 88, in _do_parse
    config, "setuptools_scm.parse_scm"
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/__init__.py", line 45, in _version_from_entrypoint
    version = _call_entrypoint_fn(config, ep.load())
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/__init__.py", line 33, in _call_entrypoint_fn
    return fn(config.absolute_root, config=config)
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/git.py", line 131, in parse
    branch=branch,
  File "/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/lib/python3.7/site-packages/setuptools_scm/version.py", line 184, in meta
    assert parsed_version is not None, "cant parse version %s" % tag
AssertionError: cant parse version V2.5.2
ERROR: InvocationError for command '/home/travis/build/melexis/sphinx-traceability-extension/.tox/sphinx-latest/bin/mlx-warnings --sphinx --maxwarnings 7 --minwarnings 7 .tox/doc_html.log' (exited with code 1)
```

No errors are thrown when using version 2.1.0. Could you give some pointers on what might cause this?

The issue in the warnings plugin project itself: https://github.com/melexis/warnings-plugin/issues/68

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,3 +1,14 @@
+v3.0.5
+
+* fix #292 - match leading 'V' character as well
+
+  https://www.python.org/dev/peps/pep-0440/#preceding-v-character
+
+v3.0.4
+=======
+
+* rerelease of 3.0.3 after fixing the release process
+
 v3.0.3  (pulled from pypi due to a packaging issue)
 ======
 

--- a/src/setuptools_scm/config.py
+++ b/src/setuptools_scm/config.py
@@ -6,7 +6,7 @@ import warnings
 
 from .utils import trace
 
-DEFAULT_TAG_REGEX = r"^(?:[\w-]+-)?(?P<version>v?\d+(?:\.\d+){0,2}[^\+]+)(?:\+.*)?$"
+DEFAULT_TAG_REGEX = r"^(?:[\w-]+-)?(?P<version>[vV]?\d+(?:\.\d+){0,2}[^\+]+)(?:\+.*)?$"
 DEFAULT_VERSION_SCHEME = "version_scheme"
 
 

--- a/testing/test_config.py
+++ b/testing/test_config.py
@@ -9,6 +9,8 @@ import pytest
         ("apache-arrow-0.9.0", "0.9.0"),
         ("arrow-0.9.0", "0.9.0"),
         ("arrow-0.9.0-rc", "0.9.0-rc"),
+        ("v1.1", "v1.1"),
+        ("V1.1", "V1.1"),
     ],
 )
 def test_tag_regex(tag, expected_version):

